src/main/java/edu/byu/cs/autograder/compile/ModuleIndependenceVerifier.java

```diff
@@ -69,4 +65,19 @@ private void checkImports(GradingContext context, StudentCodeReader reader, Set<
             }
         }
     }
+
+    private static String getPackageImport(String line) {
+        int firstSpace = line.indexOf(' ');
+        if(firstSpace == -1) return line;
```

Comment:
It's interesting that this line still returns the line for comparison against the set of packages. I suppose it will never succeed, and the comparison isn't horribly expensive because of the use of `HashSet`'s, but I would expect that if we weren't able to detect any packages in this line, that the line would be skipped and no further processing would occur.

Additionally, for those other developers who aren't intimately familiar with the problem going on here, could you provide a JavaDoc comment or even any comment that provides an overview of the kinds of patterns this is intended to match?

Otherwise, I love how carefully you've processed all this information to parse out the data we're interested in!